internal/core/application/market_price_service.go

```diff
@@ -113,9 +113,23 @@ func (m *marketPriceService) GetPrices(
 		return nil, err
 	}
 
-	marketsMap := make(map[int]domain.Market)
-	for _, v := range markets {
-		marketsMap[v.ID] = v
+	marketsMap, marketsWithSameAssetPair := groupMarkets(markets)
+	vwapPerMarket := make(map[string]decimal.Decimal)
+	if len(marketIDs) > 0 {
+		//TODO check time frame(start/stop) and average window
+		// it can happen that query is intensive, so we need to see based on
+		// time frame to scale up/down aggregation window
+		for _, v := range marketsWithSameAssetPair {
+			vwamp, err := m.marketPriceRepository.CalculateVWAP(
+				ctx, "1h", startTime, endTime, v...)
```

Comment:
I don't think we return market prices per every hour when the user asks for the prices of the whole year, nor this can work if it asks for the prices in the last hour.

Actually, maybe this is ok only for LAST_DAY time frame?

I'd like to know more about all the sub prices/balances we return for each time frame. If we have any logic, we likely have to repeat it here?